pkg/controller/accept/accept_controller.go

```diff
@@ -0,0 +1,474 @@
+			Errorf("attempt to enqueue owner that is not a cluster")
+		return
+	}
+	c.controller.enqueueCluster(cluster)
+}
+
+type jobSyncStrategy struct {
+	controller *Controller
+}
+
+func (s *jobSyncStrategy) GetOwner(key string) (metav1.Object, error) {
+	namespace, name, err := cache.SplitMetaNamespaceKey(key)
+	if err != nil {
+		return nil, err
+	}
+	if len(namespace) == 0 || len(name) == 0 {
+		return nil, fmt.Errorf("invalid key %q: either namespace or name is missing", key)
+	}
+	return s.controller.clustersLister.Clusters(namespace).Get(name)
+}
+
+func (s *jobSyncStrategy) DoesOwnerNeedProcessing(owner metav1.Object) bool {
+	cluster, ok := owner.(*clusteroperator.Cluster)
+	if !ok {
+		s.controller.logger.Warn("could not convert owner from JobSync into a cluster: %#v", owner)
+		return false
+	}
+	if cluster.Status.Accepted && cluster.Status.AcceptedJobGeneration == cluster.Generation {
+		s.controller.logger.Debugf("cluster %s/%s is already accepted for its current generation, it will not be processed", cluster.Namespace, cluster.Name)
+		return false
+	}
+
+	machineSets, err := controller.MachineSetsForCluster(cluster, s.controller.machineSetsLister)
+	if err != nil {
+		s.controller.logger.Errorf("error retrieving machine sets for cluster %s/%s: %v", cluster.Namespace, cluster.Name, err)
+		utilruntime.HandleError(fmt.Errorf("cannot retrieve machinesets for cluster %s/%s: %v", cluster.Namespace, cluster.Name, err))
+		return false
+	}
+
+	// Only process if all machinesets have been provisioned
+	totalCount := len(machineSets)
+	provisionedCount := 0
+	installedCount := 0
+	for _, ms := range machineSets {
+		if ms.Status.Provisioned {
+			provisionedCount++
+		}
+		if ms.Spec.NodeType == clusteroperator.NodeTypeMaster && ms.Status.Installed {
+			installedCount++
+		}
+	}
+
+	// With at least one machine set, all machineSets in the cluster provisioned and a
+	// master that is installed, we are ready to accept nodes
+	return totalCount > 0 && provisionedCount == totalCount && installedCount > 0
+}
+
+func (s *jobSyncStrategy) GetJobFactory(owner metav1.Object) (controller.JobFactory, error) {
+	cluster, ok := owner.(*clusteroperator.Cluster)
+	if !ok {
+		return nil, fmt.Errorf("could not convert owner from JobSync into a cluster")
+	}
+	return jobFactory(func(name string) (*v1batch.Job, *kapi.ConfigMap, error) {
+		vars, err := ansible.GenerateClusterVars(cluster)
+		if err != nil {
+			return nil, nil, err
+		}
+		job, configMap := s.controller.ansibleGenerator.GeneratePlaybookJob(
+			name,
+			&cluster.Spec.Hardware,
+			acceptPlaybook,
+			ansible.DefaultInventory,
+			vars,
+		)
+		return job, configMap, nil
+	}), nil
+}
+
+func (s *jobSyncStrategy) GetOwnerCurrentJob(owner metav1.Object) string {
+	cluster, ok := owner.(*clusteroperator.Cluster)
+	if !ok {
+		s.controller.logger.Warn("could not convert owner from JobSync into a cluster: %#v", owner)
```

Comment:
Mix of error and warn for these, error seems better.

Reply:
I think we should hold off on changing this unless we are going to change it in all of the controllers. I am not opposed to changing it in all of the controllers, but suggest we do them all at once.